pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexCombiner.java

```diff
@@ -54,7 +54,7 @@ public StarTreeIndexCombiner(File indexFile)
    */
   public Map<IndexKey, IndexValue> combine(StarTreeV2BuilderConfig builderConfig, File starTreeIndexDir)
       throws IOException {
-    Map<IndexKey, IndexValue> indexMap = new HashMap<>();
+    Map<IndexKey, IndexValue> indexMap = new TreeMap<>();
```

Comment:
Suggest making it a `List<Pair<IndexKey, IndexValue>>` instead of a `TreeMap` because we don't need to dedup

Reply:
Ack